ls1intum/Artemis
# `Development`: Add app site association for mobile app deep link support

### Checklist
#### General


#### Server

### Motivation and Context


For the iOS client we need to add an apple-app-site-association file available under /.well-known/apple-app-site-association so Deeplinks navigate into the app instead of the mobile browser. See [here](https://developer.apple.com/documentation/xcode/supporting-associated-domains) for more information. 

### Description

A static json content is served under the above mentioned route.

### Steps for Testing

1. Open the <Artemis_Instance_Base_URL>/.well-known/apple-app-site-association (e.g. http://artemis-test1.artemis.in.tum.de/.well-known/apple-app-site-association) path and inspect the json content. 
2. The content should look like this `{"applinks":{"apps":[],"details":[{"appID":"2J3C6P6X3N.de.tum.cit.artemis","paths":["/courses/*"]}]}}`
3. Open the <Artemis_Instance_Base_URL>/.well-known/assetlinks.json (e.g. http://artemis-test1.artemis.in.tum.de/.well-known/assetlinks.json) path and inspect the json content. 

### Review Progress






#### Code Review
#### Manual Tests

### Screenshots
There are no UI changes related to this PR

### Test Coverage



| Class/File | Branch | Line |
|------------|-------:|-----:|
| AppleAppSiteAssociationResource.java | 50% | 90% |
| ProgrammingExerciseRepositoryService.java | 50% | 88% |
